pages/base.page.js

```diff
@@ -0,0 +1,38 @@
+const PAGE_TIMEOUT = 10000
+
+export default class BasePage {
+
+
+   /**
+    * Abrir página
+    * @param {String} ruta a la cual acceder
+    */
+   async abrir(ruta) {
+       await browser.url(`${ruta}`);
+   }
+
+
+   /**
+    * Esperar a que un elemento sea clickeable y hacer click
+    * @param {Object} elemento a clickear
+    */
+   async clickearElemento(elemento) {
+       await elemento.waitForClickable({ timeout: PAGE_TIMEOUT });
+       await elemento.click();
+   }
+
+
+   /**
+    * Método para enviar texto a un elemento
+    * @param {Object} elemento que recibirá el texto
+    * @param {String} texto a envíar 
+    */
+   async vaciarCampoYEnviarTexto(elemento, texto){
+       await elemento.waitForClickable({ timeout: PAGE_TIMEOUT });
+       await elemento.clearValue();
+       await elemento.click();
+       await elemento.keys(texto);
```

Comment:
If you use the `setValue()` method in web elements instead of raw `keys()` you can avoid most of the code necessary for cleaning their previous value and clicking them, as it's an abstraction on top of all that.

Doing that, you can simplify it to something like:

```javascript
   async sendText(element, text){
       await element.waitForClickable({ timeout: PAGE_TIMEOUT });
       await element.setValue(text);
}
```
https://v5.webdriver.io/docs/api/element/setValue.html

Reply:
Ok, I'm learning!